### PR TITLE
Keras-like API embedding, batchnorm, padding, upsampling layers

### PR DESCRIPTION
**Pull request by hkvision**
## What changes were proposed in this pull request?

Embedding
BatchNormalization
ZeroPadding1D
ZeroPadding2D
UpSampling1D
UpSampling2D
UpSampling3D
MaxoutDense

## How was this patch tested?

Jenkins

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

**Comment by hkvision** [comment]
jenkins passed http://172.168.2.101:8080/job/BigDL-PR-Validation-v3/50/

**Comment by zhichao-li** [comment]
Please enrich the error msg with the input length: 
```+    require(input.length == 2, "Maxout only supports 2D input")
```

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/UpSampling2D.scala [hunk 1]
```diff
@@ -39,6 +40,19 @@ class UpSampling2D[T: ClassTag] (val size: Array[Int], val format: DataFormat =
...
+  override def computeOutputShape(inputShape: Shape): Shape = {
+    val input = inputShape.toSingle().toArray
+    require(input.length == 4, "UpSampling2D only supports 4D input")
+    format match {
+      case DataFormat.NCHW =>
+        Shape(input(0), input(1), input(2)*size(0), input(3)*size(1))
+      case DataFormat.NHWC =>
+        Shape(input(0), input(1)*size(0), input(2)*size(1), input(3))
+      case _ =>
+        throw new IllegalArgumentException("UpSampling2D: unsupported data format.")
```
Comment (zhichao-li) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/UpSampling2D.scala]: add the error format into the msg?

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/nn/BatchNormalizationSpec.scala [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+package com.intel.analytics.bigdl.keras.nn
+
+import com.intel.analytics.bigdl.keras.KerasBaseSpec
+import com.intel.analytics.bigdl.nn.keras.{BatchNormalization, Sequential => KSequential}
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.utils.Shape
+
+class BatchNormalizationSpec extends KerasBaseSpec {
+
+  "BatchNormalization" should "work properly" in {
```
Comment (zhichao-li) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/nn/BatchNormalizationSpec.scala]: let's add python layer for `BatchfNorm` here, then python unittest can help on this.

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/nn/EmbeddingSpec.scala [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    input(Array(1, 1)) = 1
+    input(Array(1, 2)) = 2
+    input(Array(1, 3)) = 4
+    input(Array(1, 4)) = 5
+    input(Array(2, 1)) = 4
+    input(Array(2, 2)) = 3
+    input(Array(2, 3)) = 2
+    input(Array(2, 4)) = 6
+    val output = seq.forward(input)
+    val gradInput = seq.backward(input, output)
```
Comment (zhichao-li) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/nn/EmbeddingSpec.scala]: same as batchNorm, add python layer to cover the test.

File: pyspark/bigdl/nn/layer.py [hunk 4]
```diff
@@ -605,8 +605,8 @@ class Container(Layer):
      A container usually contain some other modules which can be added through the "add" method
     '''
 
-    def __init__(self, jvalue, bigdl_type, *args):
-        super(Container, self).__init__(jvalue, bigdl_type, *args)
+    def __init__(self, jvalue, bigdl_type, is_keras=False, *args):
```
Comment (zhichao-li) [inline pyspark/bigdl/nn/layer.py]: remove is_keras here?

**Comment by zhichao-li** [comment]
LGTM

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/nn/EmbeddingSpec.scala (hunk 3 above)
Comment (hkvision) [reply spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/nn/EmbeddingSpec.scala]: Compared in python.

**Comment by hkvision** [comment]
jenkins passed http://172.168.2.101:8080/job/BigDL-PR-Validation-v3/179/
